pkg/slack/events/joblink/link.go

```diff
@@ -0,0 +1,389 @@
+				logger.Debug("Resolving path from alias.")
+				prefix := fmt.Sprintf("gs://%s", options.Bucket)
+				// dereference alias to get the path
+				var p gcs.Path
+				if err := p.Set(fmt.Sprintf("%s/%s", prefix, alias)); err != nil {
+					logger.WithError(err).Warn("Could not set path for alias read.")
+					continue
+				}
+				if p.Object() == "" {
+					logger.Warn("Alias read found empty object name.")
+					continue
+				}
+				reader, err := gcsClient.Bucket(p.Bucket()).Object(p.Object()).NewReader(context.Background())
+				if err != nil {
+					logger.WithError(err).Warn("Could not open alias for read.")
+					continue
+				}
+				symlink, err := ioutil.ReadAll(reader)
+				if err != nil {
+					logger.WithError(err).Warn("Could not read alias.")
+					continue
+				}
+				path = strings.TrimPrefix(string(symlink), prefix)
+			} else {
+				_, path, _ = gcsupload.PathsForJob(options, &dspec, "")
+			}
+			logger.WithField("path", path).Debug("Resolved full GCS path.")
+			text.WriteString("\n - Job result <https://prow.ci.openshift.org/view/gs/" + options.Bucket + "/" + path + "|link>.")
+		}
+
+		blocks = append(blocks, &slack.SectionBlock{
+			Type: slack.MBTSection,
+			Text: &slack.TextBlockObject{
+				Type: slack.MarkdownType,
+				Text: text.String(),
+			},
+		})
+	}
+	return blocks
+}
+
+type jobInfo struct {
+	// our jobs have globally unique names so we can
+	// get away with identifying them with this minimal
+	// set of information
+	Name, Id string
+	Line     int
+}
+
+// extractInfo extracts information about any jobs that were
+// linked in a comment body on Slack
+func extractInfo(event *slackevents.EventsAPIEvent) ([]*jobInfo, error) {
+	raw, ok := event.Data.(*slackevents.EventsAPICallbackEvent)
+	if !ok {
+		return nil, errors.New("could not get raw event content")
+	}
+	type blocks struct {
+		Blocks []struct {
+			Elements []struct {
+				Elements []struct {
+					Url string `json:"url"`
+				} `json:"elements"`
+			} `json:"elements"`
+		} `json:"blocks"`
+	}
+	var data blocks
+	fmt.Println(string(*raw.InnerEvent))
+	if err := json.Unmarshal(*raw.InnerEvent, &data); err != nil {
+		return nil, errors.New("could not get blocks from event data")
+	}
+	fmt.Println(data)
+
+	var infos []*jobInfo
+	for _, block := range data.Blocks {
+		for _, element := range block.Elements {
+			for _, subElement := range element.Elements {
+				if subElement.Url != "" {
+					if url, err := url.Parse(subElement.Url); err == nil {
+						infos = append(infos, infoFromUrl(url))
+					}
+				}
+			}
+		}
+	}
+
+	return infos, nil
+}
+
+func infoFromUrl(url *url.URL) *jobInfo {
+	var info jobInfo
+	switch url.Host {
+	case "prow.ci.openshift.org":
+		switch strings.Split(url.Path, "/")[1] {
+		case "job-history":
+			return infoForJobHistory(url)
+		case "log":
+			return infoForJobLog(url)
+		case "view":
+			return infoForJobView(url)
+		}
+	case "gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com":
```

Comment:
can we add this url to the api package so we don't break the tool when we give it a nicer one?